### PR TITLE
Telegram and SlackRTM were not bridging properly

### LINKED ISSUE #537: Telesync plugin behaves poorly by default w/no config
Lots of decode failures and other crap, over and over and over again.
By default, if there is no configuration, even if the plugin is loaded, it should be dormant.
Some of us enable all plugins for testing.

This is after a pip3 install -r requirements.txt --upgrade too...

```
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/local/lib/python3.5/dist-packages/telepot/asy
Mar 09 12:18:10 hangupsbot[29722]: data = yield from response.json()
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/local/lib/python3.5/dist-packages/aiohttp/cli
Mar 09 12:18:10 hangupsbot[29722]: return loads(stripped.decode(encoding))
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/lib/python3.5/json/__init__.py", line 319, in
Mar 09 12:18:10 hangupsbot[29722]: return _default_decoder.decode(s)
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/lib/python3.5/json/decoder.py", line 339, in
Mar 09 12:18:10 hangupsbot[29722]: obj, end = self.raw_decode(s, idx=_w(s, 0).end())
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/lib/python3.5/json/decoder.py", line 357, in
Mar 09 12:18:10 hangupsbot[29722]: raise JSONDecodeError("Expecting value", s, err.value) f
Mar 09 12:18:10 hangupsbot[29722]: json.decoder.JSONDecodeError: Expecting value: line 1 co
Mar 09 12:18:10 hangupsbot[29722]: During handling of the above exception, another exceptio
Mar 09 12:18:10 hangupsbot[29722]: Traceback (most recent call last):
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/local/lib/python3.5/dist-packages/telepot/asy
Mar 09 12:18:10 hangupsbot[29722]: result = yield from self.getUpdates(offset=offset, timeo
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/local/lib/python3.5/dist-packages/telepot/asy
Mar 09 12:18:10 hangupsbot[29722]: return (yield from self._parse(r))
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/local/lib/python3.5/dist-packages/telepot/asy
Mar 09 12:18:10 hangupsbot[29722]: raise BadHTTPResponse(response.status, text)
Mar 09 12:18:10 hangupsbot[29722]: telepot.exception.BadHTTPResponse: (200, '<!DOCTYPE html
Mar 09 12:18:10 hangupsbot[29722]: Bots are special Telegram accounts designed to handle me
Mar 09 12:18:10 hangupsbot[29722]: t is that you don&#39;t need to know anything about how
Mar 09 12:18:10 hangupsbot[29722]: links or username search to find your bot.</li>\n<li>Bot
Mar 09 12:18:10 hangupsbot[29722]: " style="max-height: 300px; padding: 10px 5px" /></a>\n\
```

### LINKED ISSUE #781: SlackRTM -> HObot -> Telegram fails to relay through
...but Telegram -> HObot -> Slack works just fine.

SlackRTM -> HO works.
HO -> SlackRTM works.
TG -> HO works.
HO -> TG works.

Telegram is registering on event=hangups.ChatMessageEvent (_on_hangouts_message).
Slack is registering an input handler on "allmessages".

Slack is using: `coro_send_message(hangoutid, msg, context={'slack': True}` to send messages to hangouts.

Telegram is using `coro_send_message(hangoutid, msg)` to send messages to hangouts.

I suspect that we need to add a context otherwise it won't get relayed through.

### PR DESCRIPTION
**Pull request by pleasantone**
Fix #781, also fix #537

### CODE REVIEW

**Comment by pleasantone** [comment]
@makiftasova please also take a peek... I've got a bunch more changes for TG, but we'll keep it at these for now.

**Comment by JosephBlock** [comment]
I will be making a pull request to support latest version of telepot after I finish testing.

**Comment by pleasantone** [comment]
I just did some work on tele sync, and I concur.  Actually, I want to go through and thoroughly scrub that plugin in SlackRTM to get them up to modern standards (remove all the chatmessagesegment stuff, fix all the logging).

**Comment by endofline** [comment]
I will defer code review on this until we resolve #781 since implementing it would probably mean architectural changes to all the chatbridges. Personally, I will adapt plugins syncrooms, slack to test and verify the related context relay code in #785.

**Comment by pleasantone** [comment]
Fair enough, so for v3, do we want to continue to support Slack (since you said you'd adopt it), or should I clean up SlackRTM?  I think everyone has mostly moved off the legacy plugin and over to RTM.

**Comment by endofline** [comment]
@pleasantone - oh yes, please proceed with other plugins. I'm merely trying to validate which of the 3 methods I added in that branch would be best. I suspect we may need some ancillary function for bridge-awareness with mentions/subscribe plugins and just want to get started first with some straightforward plugins.

**Comment by pleasantone** [comment]
Just to clarify, here are the tasks as I see them (in general):

* immediately fix #783 by any reasonable means necessary (either accept the reprocessor, or I'll rewrite it to use passthru and fix SlackRTM and TG).

* Come up with a set of guidelines and APIs so that we are uniform and consistent about how
we handle:
-- standardizing bridging user messages
-- standardizing bridging commands that were sent by a Hangouts user to the bot
-- standardizing accepting/bridging commands that were sent from a non-Hangouts user to the bot
-- recognizing that person A in HO is also person B in <other thing> (profile sync)
-- standardizing notifications for leave/joins between a communications entities

* give the SlackRTM plugin a good solid clean-up, remove the segment stuff, bring it up to
  modern python standards and make it readable and clean

* give the TG plugin a good solid clean-up too, there is a lot of structure weirdness that can easily
  be simplified to make the code more readable and clean

* eventually adopt the discord plugin and bring it into the fold, doing the same API cleanup

**Comment by endofline** [comment]
@pleasantone - if you're confident of the functionality of this branch, I will defer to your decision to merge as an interim fix for malfunctioning bridges in v3.0 since I do not have a functional environment to test this at the moment.

File: hangupsbot/plugins/slackrtm.py [hunk 1]
```diff
@@ -1168,22 +1179,27 @@ def handle_reply(self, reply):
...
+
         for sync in self.get_syncs(hangoutid=event.conv_id):
-            if self.sending and ': ' in event.text:
-                # this hangout message originated in slack
-                self.sending -= 1
-                command = event.text.split(': ')[1]
-                event.text = command
-                logger.debug('attempting to execute %s', command)
-                yield from self.bot._handlers.handle_command(event)
-                return
```
Comment (endofline) [inline hangupsbot/plugins/slackrtm.py]: removal of this section might be the cause of the regression in #796

Comment (endofline) [inline hangupsbot/plugins/slackrtm.py]: in retrospect, i misunderstood the nature of #796 due to inexperience with the slackrtm plugin and spectacularly uninformative issue descriptions - it appears there are two issues associated with slackrtm, and only one of them is being reported: 
* /bot` commands issued in slack are not being run - re: #796 
* `@hobot` commands are no longer recognised when entered on the slack side either

neither of these features are dear to me right now (and I'm not even sure they are gonna make the cut to v3 chatbridges) so I'm simply putting this here as an FYI

**Pull request merged**